### DISCUSSION TITLE
SSR web app Overkill ?

### DISCUSSION THREAD

**Discussion by stychu** [vercel/turborepo#1577] [Help] [answered]
Hello, I recently found about turborepo. Want to get some insight from people that use it before I can make any commitment.

I'm rebuilding a 3 years old SSR web app which is based on Node/Express/React.
I want to know if turborepo will be useful or an overkill.

The new tech stack I want to offer to my teammates to go with would be composed with these in mind: 
 - NodeJs
 - React
 - Fastify
 - Vite
 - tRPC
 - TypeScript

Given those main components and especially that `tRPC` "requires" monorepo I started looking at turborepo but Im not sure if bringing it to the table would be beneficial. 

What potential advantages would possibly be achieved using turborepo for just a Server Side Rendered React web app.

Thanks for any input.
Cheers!

**Comment by mehulkar** [comment, marked as answer]
This is just my personal opinion, but Turborepo's strengths are that it can intelligently avoid repeating work that has already been done. If you're expecting to organize your code in a way where some code will change more often than other, or if you want to define that the relationships between parts of your codebase, I think Turborepo can add value. 

It's not live yet, but @mattpocock is working on a "Why Monorepo" page for the docs also that you can preview: https://github.com/vercel/turborepo/pull/1901

Outside of that, if you want an open discussion, come hang out in the [Turborepo Discord server](https://turborepo.org/discord)!
